### ISSUE TITLE
Backprop: Add learning rate scheduling instead of random initialisation

### ISSUE THREAD

**Issue by stservice** [stSoftwareAU/NEAT-TS#1437]
## Problem

The default learning rate in `src/propagate/BackPropagation.ts` is set to `random()³`, which is heavily biased towards very small values (mean ~0.05):

```typescript
learningRate: Math.min(
  Math.max(
    options?.learningRate ??
      rng.random() * rng.random() * rng.random(),
    0.001,
  ),
  1,
),
```

A random cube gives a distribution where ~73% of values are below 0.1. When combined with generational dampening, this makes backpropagation extremely conservative and slow to converge. This is unlike memetic evolution which uses adaptive step sizing based on fitness feedback.

## Suggested Improvement

Implement proper learning rate scheduling strategies:

1. **Exponential decay**: `lr = initialLR * exp(-decayRate * epoch)`
2. **Performance-based adaptation**: Increase LR slightly when error improves, decrease when it stalls
3. **Warm restarts**: Reset LR periodically to escape local minima
4. **Sensible default**: Use a fixed reasonable default (e.g., 0.01) instead of random

At minimum, provide a more predictable and effective default learning rate that makes backpropagation competitive with other training methods.

## Files

- `src/propagate/BackPropagation.ts` (lines ~101-109)
- `src/config/TrainConfig.ts` (if learning rate schedule is added)

## Expected Outcome

More predictable and effective backpropagation convergence, reducing the gap between backprop and memetic evolution performance.
